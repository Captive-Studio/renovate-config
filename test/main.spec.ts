import { describe, test, expect } from 'vitest';
import baseJSON from '../base.json';
import applicationJSON from '../application.json';
import applicationLegacyJSON from '../applicationLegacy.json';
import libraryJSON from '../library.json';

describe('Configuration snapshot', () => {
  test('base.json', () => {
    expect(baseJSON).toMatchSnapshot();
  });
  test('application.json', () => {
    expect(applicationJSON).toMatchSnapshot();
  });
  test('applicationLegacy.json', () => {
    expect(applicationLegacyJSON).toMatchSnapshot();
  });
  test('library.json', () => {
    expect(libraryJSON).toMatchSnapshot();
  });
});
