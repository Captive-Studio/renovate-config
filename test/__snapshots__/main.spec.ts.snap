// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Configuration snapshot > application.json 1`] = `
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "description": "Default configuration for an application",
  "extends": [
    "github>Captive-Studio/renovate-config:recommended",
    ":pinDevDependencies",
  ],
}
`;

exports[`Configuration snapshot > applicationLegacy.json 1`] = `
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "description": "Default configuration for a legacy application. No automerge is done and concurrency is limited",
  "extends": [
    "github>Captive-Studio/renovate-config:base",
    "github>Captive-Studio/renovate-config//preset/groupRecommended",
    ":pinDevDependencies",
  ],
  "prConcurrentLimit": 10,
}
`;

exports[`Configuration snapshot > base.json 1`] = `
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "assigneesFromCodeOwners": false,
  "dependencyDashboard": true,
  "description": "Default base configuration for all Captive projects.",
  "extends": [
    "github>w5s/renovate-config:base",
    "github>Captive-Studio/renovate-config//preset/commit",
    "github>Captive-Studio/renovate-config//preset/schedule",
    "workarounds:all",
  ],
  "onboarding": false,
  "packageRules": [
    {
      "matchDepTypes": [
        "dependencies",
      ],
      "matchUpdateTypes": [
        "major",
      ],
      "minimumReleaseAge": "7 days",
    },
    {
      "matchDepTypes": [
        "dependencies",
      ],
      "matchUpdateTypes": [
        "minor",
      ],
      "minimumReleaseAge": "2 days",
    },
    {
      "matchDepTypes": [
        "dependencies",
      ],
      "matchUpdateTypes": [
        "patch",
      ],
      "minimumReleaseAge": "2 days",
    },
  ],
  "platformAutomerge": true,
  "prConcurrentLimit": 20,
  "prHourlyLimit": 0,
  "updateNotScheduled": false,
}
`;

exports[`Configuration snapshot > library.json 1`] = `
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "description": "Default configuration for a library",
  "extends": [
    "github>Captive-Studio/renovate-config:recommended",
    ":pinOnlyDevDependencies",
  ],
}
`;
